pkg/piecestore/pstore.go

```diff
@@ -0,0 +1,184 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package pstore
+
+import (
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/aleitner/FilePiece"
+	"github.com/zeebo/errs"
+)
+
+// Errors
+var (
+	ArgError = errs.Class("argError")
+	FSError  = errs.Class("fsError")
+)
+
+// creates datapath from hash and dir
+func PathByHash(hash, dir string) (string, error) {
+	if len(hash) < 20 {
+		return "", ArgError.New("Invalid hash length")
+	}
+
+	folder1 := string(hash[0:2])
+	folder2 := string(hash[2:4])
+	fileName := string(hash[4:])
+
+	return path.Join(dir, folder1, folder2, fileName), nil
+}
+
+/*
+	Store
+
+	Store data into piece store
+
+	hash 		(string)				Hash of the data to be stored
+	r 			(io.Reader)	        File/Stream that contains the contents of the data to be stored
+	length 	(length)				Size of the data to be stored
+	psFileOffset 	(offset)  Offset of the data that you are writing. Useful for multiple connections to split the data transfer
+	dir 		(string)				pstore directory containing all other data stored
+	returns (error) if 		  failed and nil if successful
+*/
+func Store(hash string, r io.Reader, length int64, psFileOffset int64, dir string) error {
+	if psFileOffset < 0 {
+		return ArgError.New("Offset is less than 0. Must be greater than or equal to 0")
+	}
+	if length < 0 {
+		return ArgError.New("Length is less than 0. Must be greater than or equal to 0")
+	}
+	if dir == "" {
```

Comment:
you should put these error checks into PathByHash and have them in one place

Reply:
These checks aren't related to getting the path from the hash though